### PR TITLE
3 task

### LINKED ISSUE #5: Task 3
- [x]  Create UI with ability to view all posts.
- [x]  Add ability to login for the user with a wallet(can support single wallet or use a library to support multiple wallets like [artemis](https://github.com/sonicdex/artemis)[)](https://github.com/sonicdex/artemis)))
- [x]  Add ability for logged in user to create a new post.

### PR DESCRIPTION
**Pull request by harshedabdulla**
closes #5

### CODE REVIEW

File: src/bloggy_backend/main.mo [hunk 1]
```diff
@@ -1,49 +1,79 @@
...
+  stable var postsStable: [Type.Post] = [];
+  stable var nextIdStable: Nat = 0;
+
+  // var userPrincipal: Principal = Principal.fromText("un4fu-tqaaa-aaaab-qadjq-cai");
+
+  // public func receivePrincipalId(receiverPrincipalId: Text) : async () {
+  //   userPrincipal := Principal.fromText(receiverPrincipalId);
+  //   Debug.print("Received Principal ID: " # debug_show(userPrincipal));
+  // };
+
```
Comment (mohitrajane) [inline src/bloggy_backend/main.mo]: please remove these unrequired comments

File: src/bloggy_backend/main.mo [hunk 2]
```diff
@@ -1,49 +1,79 @@
...
-  };
-  type Result<A,B> = Result.Result<A,B>;
+actor bloggy {
+  
+  
+  type Result<A, B> = Result.Result<A, B>;
 
-  var posts: [Post] = [];
+  var posts = Buffer.Buffer<Type.Post>(1);
   var nextId: Nat = 0;
```
Comment (mohitrajane) [inline src/bloggy_backend/main.mo:19]: will nextId be reset to 0 each time the contract is deployed?

File: src/bloggy_frontend/src/pages/UserLogin.jsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  useEffect(() => {
+    const textToType = 'bloggy_'
+    let index = 0
+    const interval = setInterval(() => {
+      if (index <= textToType.length) {
+        setTypedText(textToType.slice(0, index))
+        index++
+      } else {
+        clearInterval(interval)
+      }
```
Comment (mohitrajane) [inline src/bloggy_frontend/src/pages/UserLogin.jsx]: This can be done using CSS. https://www.sitepoint.com/css-typewriter-effect/

**Comment by mohitrajane** [comment]
@harshedabdulla make sure that if the user is not logged in and in `/posts`, the user should be redirected to `/login`

**Review by mohitrajane — CHANGES_REQUESTED** [review]

File: src/bloggy_backend/main.mo (hunk 2 above)
Comment (harshedabdulla) [reply src/bloggy_backend/main.mo:19]: No, nextId will not be reset to 0 each time the contract is deployed, as the upgrade process preserves its value using preupgrade and postupgrade functions.

**Pull request merged**